models/seqGAN-tensorflow/Readme.md

```diff
@@ -0,0 +1,152 @@
+  --wvpath WVPATH       Path for pretrained wordvector. Default: wordvec
+  --out_dir OUT_DIR     Output directory for test output. Default: ./output
+  --log_dir LOG_DIR     Log directory for tensorboard. Default: ./tensorboard
+  --model_dir MODEL_DIR
+                        Checkpoints directory for model. Default: ./model
+  --cache_dir CACHE_DIR
+                        Checkpoints directory for cache. Default: ./cache
+  --cpu                 Use cpu.
+  --debug               Enter debug mode (using ptvsd).
+  --cache               Use cache for speeding up load data and wordvec. (It
+                        may cause problems when you switch dataset.)
+```
+
+For hyperparameter settings, please refer to `run.py`.
+
+
+
+#### For developer
+
+- Arguments above (except ``cache``\\``debug``) are required. You should remain the same behavior (not for implementation).
+- You can add more arguments if you want.
+
+
+
+### An example of tensorboard
+
+Execute ``tensorboard --logdir=./tensorboard``, you will see the plot in tensorboard pages:
+
+Following plot are shown in this model:
+
+- gen_loss: loss when pre-training the generator.
+
+  ![](images/gen_loss.png)
+
+- gen_reward: average reward when adversarially training the generator.
+
+  ![](images/gen_rewards.png)
+
+- dis_loss: loss when pre-training and adversarially training the discriminator.
+
+  ![](images/dis_loss.png)
+
+- dis_accuracy: accuracy when pre-training and adversarially training
+
+  ![](images/dis_acc.png)
+
+
+
+
+### An example of test output
+
+Execute ``python run.py --mode test --restore best``
+
+The following output will be in `./output/[name]_test.txt`:
+
+```
+self-bleu:	0.156911
+fw_bw_bleu:	0.233483
+A polar bear walking through some greenery .
+A man throwing the frisbee towards the water .
+A man standing on a beach flying a kite .
+Two children playing with soccer players in a park .
+A man in uniform holding a tie and a white shirt .
+A cupcake made with frosting , half and <unk> on the fabric .
+A man in a cross country skiing down a hill .
+An adult bear playing with his noses on a grass covered hillside .
+A yummy food dish displayed on wooden table next to two bottles of beer .
+A man swinging a tennis racket at a volley match .
+A close up of an elephant 's <unk> forward with his plating storm .
+A brown cat sitting on top of a couch next to two keyboard on a laptop .
+A man in a suit poses with a flower festival outside of a building .
+Two giraffes and two trees standing under a tree .
+An image of cupcakes and nick on vacation .
+A plate of french fries and a sandwich .
+A woman walking down a street as her cat moves by the other side .
+An adult on personal home made with lit candles on top of a birthday cake .
+...
+```
+
+#### For developer
+
+- You should remain similar output in this task.
+
+
+
+### Performance
+
+|        | self-bleu | fwbw-bleu |
+| :----: | :-------: | :-------: |
+| MSCOCO |  0.1569   |  0.2335   |
```

Comment:
Are these results self-bleu-4 and bleu-ha-4？Maybe this metric should also support bleu-2/3.